### DISCUSSION TITLE
Applying Material Design

### DISCUSSION THREAD

**Discussion by keeday** [macgile/DataGridFilter#163] [Q&A] [answered]
Hi, and thank you for this amazing package!

I am loving it, however, due to my limited skills with styling I have some problems.

So I am using [Material Design](https://github.com/MaterialDesignInXAML/MaterialDesignInXamlToolkit) in my application.

I am trying to apply the Material Design to the DataGridFilter Headers, and I am sort-of able to apply it by doing this;

```XAML
            <fdg:FilterDataGrid x:Name="MainDataGrid" Grid.Row="0" AutoGenerateColumns="False" IsReadOnly="True" SelectionChanged="DataGrid_SelectionChanged" SelectionMode="Single" ItemsSource="{Binding LoadedNcrs}" FilterPopupBackground="White" Style="{StaticResource FilterDatagridStyle}">
                <fdg:FilterDataGrid.Resources>
                    <Style TargetType="{x:Type DataGridColumnHeader}" BasedOn="{StaticResource MaterialDesignDataGridColumnHeader}" />
                </fdg:FilterDataGrid.Resources>
                <fdg:FilterDataGrid.Columns>
                    <fdg:DataGridTextColumn Header="Record ID" Binding="{Binding ID, Mode=OneWay}" SortDirection="Descending" IsColumnFiltered="True" />
                    <fdg:DataGridTextColumn Header="Status" Binding="{Binding Status, Mode=OneWay}" IsColumnFiltered="True" />
                    <fdg:DataGridTextColumn Header="Client" Binding="{Binding ClientCode, Mode=OneWay}" IsColumnFiltered="True" />
                    <fdg:DataGridTextColumn Header="Order Number" Binding="{Binding OrderNumber, Mode=OneWay}" IsColumnFiltered="True" />
                    <fdg:DataGridTextColumn Header="Raised By" Binding="{Binding RaisedBy, Mode=OneWay}"  IsColumnFiltered="True" />
                    <fdg:DataGridTextColumn Header="Raised On" Binding="{Binding DateRaised, Mode=OneWay}" IsColumnFiltered="True" />
                </fdg:FilterDataGrid.Columns>
            </fdg:FilterDataGrid>
```
However, that results in this;

https://github.com/macgile/DataGridFilter/assets/67398250/35b2b830-05f0-4d1f-bfbd-54e4f1112173

The filter button "pushes" away the text and hides it.

I have been stuck for hours trying to figure this out, but to no avail. My skills with Styling is yet fairly limited.

Any ideas how to solve this?
I have tried to simply make the header transparent without mixing in Material Design - but that resulted in the sorting indicator to disappear.

Thanks,
Keeday

**Comment by macgile** [comment, marked as answer]
Hi,

try
```xml
<Style TargetType="{x:Type DataGridColumnHeader}" BasedOn="{StaticResource MaterialDesignDataGridColumnHeader}">
 <Setter Property="HorizontalContentAlignment" Value="Stretch" />
</Style>
```

**Comment by keeday** [comment]
> Hi,
> 
> try
> 
> ```
> <Style TargetType="{x:Type DataGridColumnHeader}" BasedOn="{StaticResource MaterialDesignDataGridColumnHeader}">
>  <Setter Property="HorizontalContentAlignment" Value="Stretch" />
> </Style>
> ```

And I sat hours trying to figure this out and it's this simple... Thank you so much! This solved the issue!
